src/scripts/main.js

```diff
@@ -1,3 +1,264 @@
 'use strict';
 
-// write your code here
+const startButton = document.querySelector('.button');
+const gameField = document.querySelector('.game-field');
+const gameScore = document.querySelector('.game-score');
+const cells = document.querySelectorAll('.field-cell');
+
+const messages = document.querySelectorAll('.message');
+const messageLose = document.querySelector('.message-lose');
+const messageWin = document.querySelector('.message-win');
+
+let score = 0;
+
+const BOARD_SIZE = 4;
+const SCORE_TO_WIN = 2048;
+
+const board = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+startButton.addEventListener('click', () => {
+  if (startButton.classList.contains('start')) {
+    startGame();
+  }
+
+  [...messages].map(message => message.classList.add('hidden'));
+
+  for (let i = 0; i < BOARD_SIZE; i++) {
+    for (let j = 0; j < BOARD_SIZE; j++) {
+      board[i][j] = 0;
+    }
+  }
+
+  [...cells].map(cell => {
+    cell.className = 'field-cell';
+    cell.textContent = '';
+  });
```

Comment:
map is used when you need to return a new array. In this case it's make sense to use forEach here.
```suggestion
  cells.forEach(cell => {
    cell.className = 'field-cell';
    cell.textContent = '';
  });
```